### PR TITLE
update torchsharp and helix image

### LINKED ISSUE #7182: Helix test fail on latest torchsharp (0.102.5) and its runtime
**System Information (please complete the following information):**
 - OS & Version: [e.g. Windows 10] 
 - ML.NET Version: [e.g. ML.NET v1.5.5]
 - .NET Version: [e.g. .NET 5.0]

**Describe the bug**

The Microsoft.ML.Torchsharp.Tests fails in the following helix tests if I update torchsharp and its runtime to 0.102.5 and 2.2.1.1.
- centos x64
- ubuntu x64

The error message from helix tests indicates some dependencies of `liblibtorchsharp` is missing. After turning on `LD_Debug`, it seems that one of the missing dependencies is `GLIBC_2.34`. Note that the image for helix test is still `centos 8 streaming` which glib version is 2.28. This could be the why the torchsharp test failures.

```bash
file=/temp/workitems/Microsoft.ML.TorchSharp.Tests/runtimes/linux-x64/native/libtorch_cpu.so
       260:
       260:     find library=libgomp-98b21ff3.so.1 [0]; searching
       260:      search path=/usr/local/lib:/usr/local/lib64            (LD_LIBRARY_PATH)
       260:       trying file=/usr/local/lib/libgomp-98b21ff3.so.1
       260:       trying file=/usr/local/lib64/libgomp-98b21ff3.so.1
       260:      search path=/temp/workitems/Microsoft.ML.TorchSharp.Tests/runtimes/linux-x64/native            (RUNPATH from file /temp/workitems/Microsoft.ML.TorchSharp.Tests/runtimes/linux-x64/native/libtorch_cpu.so)
       260:       trying file=/temp/workitems/Microsoft.ML.TorchSharp.Tests/runtimes/linux-x64/native/libgomp-98b21ff3.so.1
       260:
       260:     /lib64/libc.so.6: error: version lookup error: version `GLIBC_2.34' not found (required by /temp/workitems/Microsoft.ML.TorchSharp.Tests/runtimes/linux-x64/native/libLibTorchSharp.so) (fatal)
       260:     find library=libLibTorchSharp.so [0]; searching
       260:      search path=/usr/local/lib:/usr/local/lib64            (LD_LIBRARY_PATH)
       260:       trying file=/usr/local/lib/libLibTorchSharp.so
       260:       trying file=/usr/local/lib64/libLibTorchSharp.so
       260:      search cache=/etc/ld.so.cache
       260:      search path=/lib64/tls:/lib64:/usr/lib64/tls:/usr/lib64                (system search path)
       260:       trying file=/lib64/tls/libLibTorchSharp.so
       260:       trying file=/lib64/libLibTorchSharp.so
       260:       trying file=/usr/lib64/tls/libLibTorchSharp.so
       260:       trying file=/usr/lib64/libLibTorchSharp.so
       260:
       260:     find library=LibTorchSharp [0]; searching
       260:      search path=/usr/local/lib:/usr/local/lib64            (LD_LIBRARY_PATH)
       260:       trying file=/usr/local/lib/LibTorchSharp
       260:       trying file=/usr/local/lib64/LibTorchSharp
       260:      search cache=/etc/ld.so.cache
       260:      search path=/lib64/tls:/lib64:/usr/lib64/tls:/usr/lib64                (system search path)
       260:       trying file=/lib64/tls/LibTorchSharp
       260:       trying file=/lib64/LibTorchSharp
       260:       trying file=/usr/lib64/tls/LibTorchSharp
       260:       trying file=/usr/lib64/LibTorchSharp
       260:
       260:     find library=libLibTorchSharp [0]; searching
       260:      search path=/usr/local/lib:/usr/local/lib64            (LD_LIBRARY_PATH)
       260:       trying file=/usr/local/lib/libLibTorchSharp
       260:       trying file=/usr/local/lib64/libLibTorchSharp
       260:      search cache=/etc/ld.so.cache
       260:      search path=/lib64/tls:/lib64:/usr/lib64/tls:/usr/lib64                (system search path)
       260:       trying file=/lib64/tls/libLibTorchSharp
       260:       trying file=/lib64/libLibTorchSharp
       260:       trying file=/usr/lib64/tls/libLibTorchSharp
       260:       trying file=/usr/lib64/libLibTorchSharp
```

## Solution
- on mlnet side: update helix test image from centos-streaming8-mlnet-helix and ubuntu-18.04-mlnet-helix to centos-streaming9-mlnet-helix and ubuntu-22.04-mlnet-helix
- on [dotnet-buildtools-prereqs-docker](https://github.com/dotnet/dotnet-buildtools-prereqs-docker) side: add centos-streaming9-mlnet-helix. PR:https://github.com/dotnet/dotnet-buildtools-prereqs-docker/pull/1130
- on torchsharp side: downgrade the building image from ubuntu-22.04-mlnet to ubuntu-20.04 to resolve glibc incompatible issue. PR:https://github.com/dotnet/TorchSharp/pull/1338

### PR DESCRIPTION
**Pull request by LittleLittleCloud**
We are excited to review your PR.

So we can do the best job, please check:

- [ ] There's a descriptive title that will make sense to other developers some time from now. 
- [ ] There's associated issues. All PR's should have issue(s) associated - unless a trivial self-evident change such as fixing a typo. You can use the format `Fixes #nnnn` in your description to cause GitHub to automatically close the issue(s) when your PR is merged.
- [ ] Your change description explains what the change does, why you chose your approach, and anything else that reviewers should know.
- [ ] You have included any necessary tests in the same PR.

fix #7182

### CODE REVIEW

File: src/Microsoft.ML.GenAI.Core/Microsoft.ML.GenAI.Core.csproj [hunk 1]
```diff
@@ -9,10 +9,7 @@
 
   <ItemGroup>
     <PackageReference Include="System.Memory" Version="$(SystemMemoryVersion)" />
-    <PackageReference Include="TorchSharp" Version="0.102.5" />
-    <PackageReference Include="libtorch-cpu-win-x64" Version="2.2.1.1" Condition="$([MSBuild]::IsOSPlatform('Windows'))" PrivateAssets="all" />
-    <PackageReference Include="libtorch-cpu-linux-x64" Version="2.2.1.1" Condition="$([MSBuild]::IsOSPlatform('Linux'))" PrivateAssets="all" />
-    <PackageReference Include="libtorch-cpu-osx-x64" Version="2.2.1.1" Condition="$([MSBuild]::IsOSPlatform('OSX'))" PrivateAssets="all" />
+    <PackageReference Include="TorchSharp" Version="$(TorchSharpVersion)" />
```
Comment (michaelgsharp) [inline src/Microsoft.ML.GenAI.Core/Microsoft.ML.GenAI.Core.csproj:12]: We should revert this change for disk space reasons. I mean update the version yes, but keep the cpu libs split out by OS

Comment (LittleLittleCloud) [reply src/Microsoft.ML.GenAI.Core/Microsoft.ML.GenAI.Core.csproj:12]: This is not torchsharp runtime package, so the size will not be huge

It would be upon user’s preference to choose which torch runtime to install when they consume genai packages

Comment (michaelgsharp) [reply src/Microsoft.ML.GenAI.Core/Microsoft.ML.GenAI.Core.csproj:12]: Gotcha. Sounds good to me then as is.
